.travis.yml

```diff
@@ -40,8 +41,22 @@ jobs:
       install:
         - "cd src && npm install && cd -"
       script:
-        - "docker build -t etherpad:test ."
-        - "docker run -d -p 9001:9001 etherpad:test && sleep 3"
+        - "docker build -t ${IMAGE_NAME} ."
+        - "docker run -d -p 9001:9001 ${IMAGE_NAME} && sleep 3"
+        - "cd src && npm run test-container"
+    - name: "Test a docker-compose with MySQL"
+      install:
+        - "cd src && npm install && cd -"
+      script:
+        - "docker-compose -f docker-compose_mysql.yml build"
+        - "docker-compose -f docker-compose_mysql.yml up -d && sleep 60"
```

Comment:
Are we waiting for the DB to come up here? If so, is there a way to wait for some event, like - for example - a TCP port that starts accepting connections?

Hardcoded timeouts always make my scars aching again :smile:

Reply:
> Are we waiting for the DB to come up here?

Yes, that's what is done here.

> If so, is there a way to wait for some event, like - for example - a TCP port that starts accepting connections?

Made it the same way the "_Test the Dockerfile_" job waits for Etherpad to be ready, but I guess it should be possible using custom netcat commands.

The best though would be to add the wait directly into the Dockerfile, so that Etherpad waits for the DB to be ready before starting.
This is something I've done in our fork through a custom simple script [Monogramm/docker-etherpad wait-for-service.sh](https://github.com/Monogramm/docker-etherpad/blob/master/template/wait-for-service.sh) but I was not sure if you were interested.
Also wondering if using the standard [wait-for-it](https://github.com/vishnubob/wait-for-it) might be better or not here.

Reply:
> Made it the same way the "Test the Dockerfile" job waits for Etherpad to be ready

And it was a wise decision. Sticking to whatever a project is doing is a sign of maturity, thanks. In this case, a better way exists and we are deciding to use it.

> but I guess it should be possible using custom netcat commands.

Wow, I think I've done something similar in commit 4a17443a2e6a:
https://github.com/ether/etherpad-lite/blob/5503ebdb949eeb9e4058ee5e3e072436a14bc255/tests/frontend/travis/runner.sh#L23-L31

But I did not know of the existence of more engineered solutions.

About what to use (something in the Dockerfile, or a custom command invoked from `.travis.yml`, I'll only offer two remarks:
- a dockerised Etherpad can use various DB backends. Waiting for an external process to be ready may not be necesary in every case (think about dirtydb or sqlite).
- if possible, let's choose a single mechanism to wait for an open port where applicable (it may be a custom one or [wait-for-it](https://github.com/vishnubob/wait-for-it). Can we backport that mechanism to `ests/frontend/travis/runner.sh`, too?